### PR TITLE
Add cellular networking tests.

### PR DESCRIPTION
**Pull request by petedyerarm**
Added pytest files to control the cellular module and test the cellular
behaviour.

Converted the enable-wifi step from shell to pytest.

Added python file for communicating with the cellular module over the
serial port (using pyserial)

Added installation of pyserial on dut and enable wifi test

### CODE REVIEW

**Comment by petedyerarm** [comment]
synchronize with https://github.com/ARMmbed/mbl-tools/pull/360

File: ci/lava/tests/controlCellular.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python3
```
Comment (diegorusso) [inline ci/lava/tests/controlCellular.py]: Can we rename the file to `control_cellular.py`?

File: ci/lava/tests/controlCellular.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+    parser.add_argument(
+        "command",
+        help="",
+        choices=["comms", "status", "activate", "deactivate", "power"],
+    )
+    return parser
+
+
+def comms_check():
```
Comment (diegorusso) [inline ci/lava/tests/controlCellular.py]: Can we create a class to group the following functions?

File: ci/lava/tests/test-cellular-and-wifi-dut.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+import time
+import controlCellular as cellular
+
+
+class TestRunner:
+    """Class to encapsulate the testing of cellular connectivity on a DUT."""
+
+    overall_result = "fail"
+
+    def _run_ping_tests(self, interface, expected, execute_helper):
```
Comment (diegorusso) [inline ci/lava/tests/test-cellular-and-wifi-dut.py]: Can we move all the private methods at the end of the class?

File: ci/lava/tests/test-cellular-and-wifi-dut.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+import pytest
+import re
+import random
+import string
+import time
+import controlCellular as cellular
+
+
+class TestRunner:
```
Comment (diegorusso) [inline ci/lava/tests/test-cellular-and-wifi-dut.py]: Can we have a better name of the class?

File: ci/lava/tests/test-cellular-and-wifi-host.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+The test_component function runs pytest on the DUT via the mbl-cli on the
+copied files.
+
+"""
+import os
+import pytest
+import re
+
+
+class TestRunner:
```
Comment (diegorusso) [inline ci/lava/tests/test-cellular-and-wifi-host.py]: Same as above

**Comment by diegorusso** [comment]
Please fix the sanity checks.

File: ci/lava/dependencies/test-create-python_environment-on-device.py [hunk 6]
```diff
@@ -21,7 +21,23 @@
 class Test_Create_Pytest_Environment_On_DUT:
```
Comment (diegorusso) [inline ci/lava/dependencies/test-create-python_environment-on-device.py]:
```suggestion
class TestCreatePytestEnvironmentOnDUT:
```

File: ci/lava/dependencies/test-enable-wifi.py [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+import pytest
+import re
+import time
+
+QCADriverUrl = (
+    "https://www.nxp.com/lgfiles/NMG/MAD/YOCTO/firmware-qca-2.0.3.bin"
+)
+
+
+class Test_Enable_Wifi:
```
Comment (diegorusso) [inline ci/lava/dependencies/test-enable-wifi.py]:
```suggestion
class TestEnableWiFi:
```

File: ci/lava/dependencies/test-create-python_environment-on-device.py (hunk 6 above)
Comment (petedyerarm) [reply ci/lava/dependencies/test-create-python_environment-on-device.py]: Changed to TestCreatePythonEnvironmentOnDUT

**Pull request merged**